core/src/test/scala/integration/kafka/api/BaseProducerSendTest.scala

```diff
@@ -302,6 +298,70 @@ abstract class BaseProducerSendTest extends KafkaServerTestHarness {
   }
 
   /**
+    * Checks partitioning behavior before and after partitions are added
+    *
+    * Producer will attempt to send messages to the partition specified in each record, and should
+    * succeed as long as the partition is included in the metadata.
+    */
+  @Test
+  def testSendBeforeAndAfterPartitionExpansion() {
+    val producer = createProducer(brokerList)
+
+    // create topic
+    TestUtils.createTopic(zkUtils, topic, 1, 2, servers)
+    val partition0 = 0
+
+    var futures0 = (1 to numRecords).map { i =>
+      producer.send(new ProducerRecord(topic, partition0, null, ("value" + i).getBytes))
+    }.map(_.get(30, TimeUnit.SECONDS)).toList
```

Comment:
do you need `toList` here and the other similar code? I would have thought that you don't.